ethereum_history_api/circuits/lib/src/verifiers/transaction_test.nr

```diff
@@ -0,0 +1,52 @@
+mod assert_tx_proof {
+    use crate::verifiers::transaction::assert_tx_proof;
+    use crate::fixtures::mainnet::{
+        cancun::small_block::{header::transactions_root, transaction_proof::proof},
+        homestead::fork::{header::transactions_root as legacy_transactions_root, transaction_proof::proof as legacy_proof}
+    };
+    use crate::misc::arrays::alter_array;
+
+    #[test]
+    fn success() {
+        assert_tx_proof(proof, transactions_root);
+    }
+
+    #[test]
+    fn legacy_tx() {
+        assert_tx_proof(legacy_proof, legacy_transactions_root);
+    }
+
+    #[test(should_fail_with="Internal node hash does not match the hash extracted from the preceding node")]
+    fn invalid_tx_root() {
+        assert_tx_proof(proof, legacy_transactions_root);
+    }
+
+    #[test(should_fail_with="Arrays do not match at specified offset")]
+    fn invalid_value() {
+        let mut proof_invalid_value = proof;
+        proof_invalid_value.value[proof_invalid_value.value.len() - 1] += 1;
+        assert_tx_proof(proof_invalid_value, transactions_root);
+    }
+
+    #[test(should_fail_with="Internal node hash does not match the hash extracted from the preceding node")]
+    fn invalid_proof() {
+        let mut proof_invalid_proof = proof;
+        proof_invalid_proof.proof[0] += 1;
+        assert_tx_proof(proof_invalid_proof, transactions_root);
+    }
+
+    #[test(should_fail_with="Transaction root: leaf node hash does not match the hash extracted from the preceding node")]
+    fn invalid_key() {
+        let mut proof_invalid_key = proof;
+        proof_invalid_key.key[0] += 1;
+        assert_tx_proof(proof_invalid_key, transactions_root);
+    }
+
+    // TODO: This test should fail with an error message
```

Comment:
Leftover TODO

Reply:
No? There is still no proper assertion message:
```rust
// TODO: This test should fail with an error message
    #[test(should_fail)]
    fn invalid_depth() {
```